### PR TITLE
Kim CNN OOP Refactoring

### PR DESCRIPTION
**Pull request by tuzhucheng**
Refactor Kim CNN to use OOP (similar to MP-CNN and VDPWI).

Trainer and evaluator are moved to their own classes. The core training algorithm remains unchanged, it is just structured differently.

### CODE REVIEW

**Comment by tuzhucheng** [comment]
Updated README with SST-1 accuracy after hyperparameter tuning per @Impavidity's request.

| Test Accuracy on SST-1         |    rand    |    static    |    non-static  |  multichannel   |
|:------------------------------:|:----------:|:------------:|:--------------:|:---------------:|
| Paper                          |    45.0    |     45.5     |      48.0      |      47.4       |
| PyTorch using above configs    |    41.5    |     44.7     |      47.4      |      47.5       | 

With the exception of `rand`, other modes can get within 1% (absolute) accuracy.
